harness/determined/profiler.py

```diff
@@ -0,0 +1,524 @@
+            if time.time() - last_measurement_time > self.MEASUREMENT_INTERVAL:
+                immutable_batch_idx = self.current_batch_idx
+                cpu_util_measurement = cpu_util_collector.measure(immutable_batch_idx)
+                gpu_util_measurements = gpu_util_collector.measure(immutable_batch_idx)
+                net_thru_sent_measurement, net_thru_recv_measurement = network_throughput_collector.measure(immutable_batch_idx)
+                free_memory_measurement = free_memory_collector.measure(immutable_batch_idx)
+                disk_read_thru_measurement, disk_write_thru_measurement, iops_measurement = disk_collector.measure(immutable_batch_idx)
+
+                for gpu_uuid in gpu_util_measurements.keys():
+                    self.current_batch.add_gpu_measurement(SysMetricType.GPU_UTIL_METRIC, gpu_uuid, gpu_util_measurements[gpu_uuid])
+
+                self.current_batch.add_nongpu_measurement(SysMetricType.NET_THRU_SENT_METRIC, net_thru_sent_measurement)
+                self.current_batch.add_nongpu_measurement(SysMetricType.NET_THRU_RECV_METRIC, net_thru_recv_measurement)
+                self.current_batch.add_nongpu_measurement(SysMetricType.DISK_IOPS_METRIC, iops_measurement)
+                self.current_batch.add_nongpu_measurement(SysMetricType.DISK_THRU_READ_METRIC, disk_read_thru_measurement)
+                self.current_batch.add_nongpu_measurement(SysMetricType.DISK_THRU_WRITE_METRIC, disk_write_thru_measurement)
+                self.current_batch.add_nongpu_measurement(SysMetricType.FREE_MEM_METRIC, free_memory_measurement)
+                self.current_batch.add_nongpu_measurement(SysMetricType.SIMPLE_CPU_UTIL_METRIC, cpu_util_measurement)
+
+                last_measurement_time = time.time()
+
+
+            # Check if it is time to flush the batch and start a new batch
+            if time.time() - batch_start_time > self.FLUSH_INTERVAL:
+                self.enqueue_for_async_send(self.current_batch.convert_to_post_format())
+                self.current_batch.clear()
+                batch_start_time = time.time()
+
+            time.sleep(0.02)
+
+
+
+
+
+
+
+
+
+# This is a thread that exists solely so that we can make API calls without blocking
+# It has a Queue through which work is sent to the thread
+class ProfilerSenderThread(threading.Thread):
+    POLL_INTERVAL_SECS = 0.5
+    def __init__(self, inbound_queue: queue.Queue, master_url: str) -> None:
+        self.master_url = master_url
+        self.inbound_queue = inbound_queue
+        self.quitting = False
+        super().__init__()
+
+    def run(self) -> None:
+        while True:
+            if self.quitting:
+                break
+
+            try:
+                batch_to_send = self.inbound_queue.get_nowait()
+            except queue.Empty:
+                time.sleep(ProfilerSenderThread.POLL_INTERVAL_SECS)
+                continue
+
+            self.send_batch(batch_to_send)
```

Comment:
It seems like something like I listed below would work just fine for this.
```suggestion
            self.send_batch(self.inbound_queue.get(timeout=SomethingReasonableSoWeDontBlockForever))
```